subsys/bluetooth/audio/ascs.c

```diff
@@ -1264,7 +1269,7 @@ static int ascs_ep_set_codec(struct bt_audio_ep *ep, uint8_t id, uint16_t cid,
 		LOG_ERR("Unable to parse Codec Config: len %u", ad.len);
 		(void)memset(codec, 0, sizeof(*codec));
 
-		return -EINVAL;
+		return BT_ASCS_REASON_CODEC_DATA;
```

Comment:
Ditto with the change here: If we go away from the typical `< 0 => Error, 0 = Success, > 0 => success with a additional info`, we should perhaps just use `uint8_t` and only use `BT_ASCS_REASON_` as return values, ideally with using an `enum` to make it clearer